UnitySDK/Assets/ML-Agents/Scripts/DemonstrationRecorder.cs

```diff
@@ -15,24 +15,37 @@ public class DemonstrationRecorder : MonoBehaviour
         private string filePath;
         private DemonstrationStore demoStore;
 
-        /// <summary>
-        /// Initializes Demonstration store.
-        /// </summary>
         private void Start()
         {
             if (Application.isEditor && record)
             {
-                recordingAgent = GetComponent<Agent>();
-                demoStore = new DemonstrationStore();
-                demonstrationName = SanitizeName(demonstrationName);
-                demoStore.Initialize(
-                    demonstrationName, 
-                    recordingAgent.brain.brainParameters, 
-                    recordingAgent.brain.name);            
-                Monitor.Log("Recording Demonstration of Agent: ", recordingAgent.name);
+                InitializeDemoStore();
+            }
+        }
+
+        private void Update()
+        {
+            if (Application.isEditor && record && demoStore == null)
+            {
+                InitializeDemoStore();
             }
         }
 
+        /// <summary>
+        /// Creates demonstration store for use in recording.
+        /// </summary>
+        private void InitializeDemoStore()
+        {
+            recordingAgent = GetComponent<Agent>();
```

Comment:
What if this is null ?

Reply:
The component requires an agent see line 9 `[RequireComponent(typeof(Agent))]`